File: spec/support/single_object_dvd/linux/interfaces/perl/IpLocatorSample.pl

#********************************************************************************
#If you should have any comments, suggestions or improvements to these samples, 
#we welcome you to contact us at SampleCode@melissadata.com also please visit our 
#developers bulletin board at forum.melissadata.com.
#********************************************************************************

use mdIpLocatorPerlWrapper;


my $LICENSE = "License Here";
my $FILELOC = "/DQT/Current";

#.................................................................................


$ipObj = new mdIpLocatorPerlWrapper::mdIpLocator();

# ********************** LICENSE STRINGS ***********************
# *     To unlock the full functionality of IpLocator Object,    *
# * please call a sales representative at 1-800-MELISSA ext. 3 * 
# *         (1-800-635-4772 x3) for a license string.          *
# *      Without a valid license string, IpLocatorCheck will     *
# *                	not function.      		         *
# **************************************************************
        
# Set the License String 
$ipObj->SetLicenseString($LICENSE);


# Set paths for data files in Initialize method
$ipObj->SetPathToIpLocatorFiles($FILELOC);

my $InitResult = $ipObj->InitializeDataFiles();
if ($InitResult != 0)
{
	printf $ipObj->GetInitializeErrorString(), "\n";
	die "Cannot Initialize IpLocator Object";	
}

#.................................................................................

# Get the Build number and DB Date

print "========================================================================\n";
print "Build Number: ", $ipObj->GetBuildNumber(), "\n";
print "Initialize Status: ", $ipObj->GetInitializeErrorString(), "\n";
print "========================================================================\n";

print "Database Date: ", $ipObj->GetDatabaseDate(), "\n";
print "Database Expiration Date: ", $ipObj->GetDatabaseExpirationDate(), "\n";
print "License Expiration Date: ", $ipObj->GetLicenseStringExpirationDate(), "\n";

#.................................................................................
$ipaddress = "";
while(substr($ipaddress,0,1) ne "X")
{
	print "Enter IP Address (X to quit):";
	$ipaddress = <>;
	$ipObj->LocateIpAddress($ipaddress);

	if(index($ipObj->GetResults(),"IS01")> -1)
	{

		print "Latitude: " , $ipObj->GetLatitude() , "\n";
		print "Longitude: ", $ipObj->GetLongitude() , "\n";
		print "City: " , $ipObj->GetCity() , "\n";
		print "Region: " , $ipObj->GetRegion() , "\n";	
		print "Zip: " , $ipObj->GetZip() , "\n";
      	print "Country: " , $ipObj->GetCountry() , "\n";
      	print "Country Abbreviation: " , $ipObj->GetCountryAbbreviation() , "\n";
 	     print "Domain Name: " , $ipObj->GetDomainName() , "\n";
		print "ISP: " , $ipObj->GetISP() , "\n";
		}
	elsif(index($ipObj->GetResults(),"IS02")> -1)
	{
		print "IP Address Unknown. \n";
	}
	elsif(index($ipObj->GetResults(),"IE01")> -1)
	{
		print "IP Address Empty or Not Well Formed. \n";
	}

	print "========================================================================\n";
}

File: spec/support/single_object_dvd/linux/interfaces/java/IpLocatorSample.java
/********************************************************************************
If you should have any comments, suggestions or improvements to these samples,
we welcome you to contact us at SampleCode@melissadata.com also please visit our
developers bulletin board at forum.melissadata.com.
********************************************************************************/

import com.melissadata.*;
import java.io.*;

public class IpLocatorSample {

  public static String the_license = "License Here";
  public static String datafile_dir = "/DQT/Current";

  public static void main(String args[]) throws IOException 
  {
		Boolean i;

		mdIpLocator ip = new mdIpLocator();

		// ************************************************************//
		// To unlock the full functionality of IpLocator Object,
		// please call a sales representative at 1-800-MELISSA ext. 3
		// You will need a valid license string to use IpLocator Object
		// outside.
		// ************************************************************//

		/* Set License and Datafile path */
		ip.SetLicenseString(the_license);
		ip.SetPathToIpLocatorFiles(datafile_dir);

		//Initialize Data Files
		mdIpLocator.ProgramStatus result = ip.InitializeDataFiles();

		//Print Initialization Status Results
		System.out.println("*****  Initialization Info  *****");
		System.out.println("Build number: " + ip.GetBuildNumber());
		System.out.println("Database Date: " + ip.GetDatabaseDate());
		System.out.println("Database Expiration Date: " + ip.GetDatabaseExpirationDate());
		System.out.println("License Expiration Date: " + ip.GetLicenseStringExpirationDate());

		if (result != mdIpLocator.ProgramStatus.ErrorNone)
		{
			//Problem during initialization
			System.out.println("Initialization Error: " + ip.GetInitializeErrorString());
			return;
		}
		
		String input = "";
		BufferedReader stdin = new BufferedReader( new InputStreamReader( System.in ) );
		while (!input.equals("X"))
		{

			//Get Input
			System.out.print("Enter IP Address: ");
			ip.LocateIpAddress(stdin.readLine());

			if (ip.GetResults().contains("IS01"))
                {
                    System.out.println("Latitude: " + ip.GetLatitude());
                    System.out.println("Longitude: " + ip.GetLongitude());
                    System.out.println("City: " + ip.GetCity());
                    System.out.println("Region: " + ip.GetRegion());
                    System.out.println("Zip: " + ip.GetZip());
                    System.out.println("Country: " + ip.GetCountry());
                    System.out.println("Country Abbreviation: " + ip.GetCountryAbbreviation());
                    System.out.println("Domain Name: " + ip.GetDomainName());
                    System.out.println("ISP: " + ip.GetISP());
                }
                else
                {
                    if(ip.GetResults().contains("IS02"))
			  {
			  	System.out.println("IP Address Unknown");
			  }	
			  else if(ip.GetResults().contains("IE01"))
			  {
			  	System.out.println("IP Address Empty or Not Well Formed");
			  }
                }


			System.out.print("\nPress Enter to Continue, X to quit:");
			input = stdin.readLine();
		}	
		ip.delete();
  }
}
